pkg/vaulturl/vaulturl_client.go

```diff
@@ -0,0 +1,17 @@
+package vaulturl
```

Comment:
I am wondering if we should name this `secreturl` instead. it is something more generic than vault. In my mind, vault is just one implementation of this interface.

Reply:
good call!

Reply:
then I guess the URLs need to be `secret:` too?

Reply:
sgtm

Reply:
ok refactoring to `secreturl` package but gonna stick with `vault:` and `local:` URLs